WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanFragment.kt

```diff
@@ -125,10 +125,16 @@ class ScanFragment : Fragment(R.layout.scan_fragment) {
 
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         super.onActivityResult(requestCode, resultCode, data)
-        if (requestCode == RequestCodes.SHOW_THREAT_DETAILS &&
-            data?.getBooleanExtra(ThreatDetailsFragment.REQUEST_SCAN_STATE, false) == true
-        ) {
-            viewModel.onScanStateRequested()
+        if (requestCode == RequestCodes.SHOW_THREAT_DETAILS) {
+            data?.let {
+                val threatId = it.getLongExtra(ThreatDetailsFragment.REQUEST_FIX_STATE, 0L)
+                val messageRes = it.getIntExtra(ThreatDetailsFragment.REQUEST_SCAN_STATE, 0)
+                if (threatId > 0L) {
+                    viewModel.onFixStateRequested(threatId)
+                } else {
+                    viewModel.onScanStateRequestedWithMessage(messageRes)
```

Comment:
⚠️ The `else` will be performed even when `data?.getBooleanExtra(ThreatDetailsFragment.REQUEST_SCAN_STATE, false) == false` (messageRes == 0) -> is this change intended?

Reply:
No that's not intended. Thanks for catching it 🙇‍♀️. 
Add if check in 562ea14